packages/@tailwindcss-cli/src/commands/build/index.ts

```diff
@@ -218,15 +215,28 @@ export async function handle(args: Result<ReturnType<typeof options>>) {
             args['--input'] ?? base,
           )
 
-          build = compile(input).build
-          compiledCss = build(candidates)
+          // Create a new compiler, given the new `input`
+          compiler = compile(input)
+
+          // Re-scan the directory to get the new `candidates`
+          scanDirResult = scanDir({ base, contentPaths: compiler.globs, outputGlobs: true })
+
+          // Setup new watchers
+          cleanupWatchers = await createWatchers(scanDirResult.globs, handle)
```

Comment:
We need to know the new globs before we can setup the new watchers. This also means that we technically have a potential race condition if a file changed between these 2 function calls.

One solution could be to keep the previous watchers alive and re-trigger a change when new files are created or existing files updated. Once the new watchers are created, we could cleanup the old ones.

Another solution could be to run `scanDir` _again_ once the watchers are setup. That way we will have all the missing candidates if a new file was added in the meantime.


Is this something we should try to figure out already? 🤔

Reply:
Hmm, probably? I feel like reliability in the dev setup is super important. 

> Another solution could be to run `scanDir` again once the watchers are setup.

I like this approach. It shouldn't yield files that are invalid and since we check modification date on the rust side this should no-op for the most part, right?

Reply:
Note: For the initial build, we made sure to setup the watchers before calling the actual `compiler.build()` step to ensure that we don't miss any changes in the meantime.

For this particular case, this re-creating all the watchers step will only ever happen if the CSS (or any of its imported CSS files) changes. I don't think it's worth to try and solve this issue until we have proof that somebody runs into this issue.

I think it will only happen if you have a CSS file, you change it (but don't save yet), and you have a new unsaved file that will match any of the globs in the CSS file. Then you use a "save all" feature in your editor. If then the save of the new file is delayed just long enough, then we could run into this issue, maybe.

(Discussing this with @philipp-spiess on a call right now. Writing this for future reference purposes)